src/workers/newView.ts

```diff
@@ -56,56 +56,62 @@ RETURNING "currentStreak", "maxStreak"
 const incrementReadingStreak = async (
   con: DataSource,
   data: DeepPartial<View>,
-): Promise<boolean> => {
-  const users = con.getRepository(User);
-  const repo = con.getRepository(UserStreak);
-  const { userId, timestamp } = data;
+): Promise<boolean> =>
+  con.transaction(async (entityManager) => {
+    const users = entityManager.getRepository(User);
+    const repo = entityManager.getRepository(UserStreak);
+    const { userId, timestamp } = data;
 
-  if (!userId) {
-    return false;
-  }
+    if (!userId) {
+      return false;
+    }
 
-  const viewTime = timestamp ? new Date(timestamp as string) : new Date();
+    const viewTime = timestamp ? new Date(timestamp as string) : new Date();
 
-  // TODO: This code is temporary here for now because we didn't run the migration yet,
-  // so not every user is going to have a row in the user_streak table and we need to create it.
-  // We can get rid of it once/if we implement the migration in https://dailydotdev.atlassian.net/browse/MI-70
-  const user = await users.findOne({
-    where: { id: userId },
-    relations: ['streak'],
-  });
-  const streak = await user?.streak;
-  if (user && !streak) {
-    await repo.save(
-      repo.create({
-        userId,
-        currentStreak: 1,
-        totalStreak: 1,
-        maxStreak: 1,
-        lastViewAt: viewTime,
-      }),
-    );
-  } else {
-    const results = await repo.query(INC_STREAK_QUERY, [
+    // TODO: This code is temporary here for now because we didn't run the migration yet,
+    // so not every user is going to have a row in the user_streak table and we need to create it.
+    // We can get rid of it once/if we implement the migration in https://dailydotdev.atlassian.net/browse/MI-70
+    const user = await users.findOne({
+      where: { id: userId },
+      relations: ['streak'],
+    });
+
+    const streak = await user?.streak;
+    // we need to check before we actually update the max streak
+    // otherwise we end up in a spot where we create an alert record
+    // every time a post is viewed, even if we already alerted the user
+    const { currentStreak, maxStreak } = streak || {};
+
+    // milestones are currently defined on fibonacci sequence
+    const showStreakMilestone =
+      isFibonacci(currentStreak) &&
+      streak?.lastViewAt?.getDate() !== viewTime.getDate();
+
+    const showMaxStreakMilestone =
+      streak &&
+      currentStreak + 1 > maxStreak &&
```

Comment:
Why is this current streak + 1?

Reply:
We are checking if the next increment is a new max streak here because this check is happening before we actually perform the query to increment the value in the db